src/main/java/com/surrealdb/config/SurrealCrudRepositoryImpl.java

```diff
@@ -0,0 +1,142 @@
+            throw new MissingSurrealTableAnnotationException(entityType);
+        }
+        return entityType.getAnnotation(SurrealTable.class).value();
+    }
+
+    @Override
+    public Optional<T> findById(ID id) {
+        String query = String.format("select * from %s where id = '%s';", tableName, id.toString());
+        List<QueryResult<T>> response = driver.query(query, Collections.emptyMap(), entityType);
+        if (response == null || response.isEmpty()) {
+            return Optional.empty();
+        }
+        List<T> res = response.stream()
+                .flatMap(qr -> qr.getResult().stream())
+                .toList();
+        if (!res.isEmpty()) {
+            return Optional.ofNullable(res.get(0));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<T> findAll() {
+        String query = String.format("select * from %s;", tableName);
+        List<QueryResult<T>> response = driver.query(query, Collections.emptyMap(), entityType);
+        if (response == null || response.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return response.stream()
+                .flatMap(qr -> qr.getResult().stream())
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public T save(T entity) {
+        processSequenceIds(entity);
+        return driver.create(tableName, entity);
+    }
+
+    private void processSequenceIds(T entity) {
+        for (Field field : entity.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(SurrealSequenceId.class)) {
+                field.setAccessible(true);
+                try {
+                    Object currentValue = field.get(entity);
+                    if (currentValue == null) {
+                        String sequenceName = field.getAnnotation(SurrealSequenceId.class).sequenceName();
+                        Long nextIdValue = generateRouteId(sequenceName);
+                        if (field.getType().equals(String.class)) {
+                            field.set(entity, String.valueOf(nextIdValue));
+                        } else if (field.getType().equals(Long.class) || field.getType().equals(long.class)) {
+                            field.set(entity, nextIdValue);
+                        } else if (field.getType().equals(Integer.class) || field.getType().equals(int.class)) {
+                            field.set(entity, nextIdValue.intValue());
+                        } else {
+                            throw new IllegalArgumentException("Unsupported field type for @SurrealSequenceId: " + field.getType());
+                        }
+                    }
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Failed to access or set value for @SurrealSequenceId field", e);
+                }
+            }
+        }
+    }
+
+    private Long generateRouteId(String sequenceName) {
```

Comment:
I don't know what this is but it looks wrong? SurrealDB doesnt provide sequences and creating them artificially is misleading, incorrect, and prone to unexpected bugs.